whitelist/src/lib.rs

```diff
@@ -0,0 +1,335 @@
+    pub fn add_staking_pool(&mut self, staking_pool_account_id: AccountId) -> bool {
+        assert!(
+            env::is_valid_account_id(staking_pool_account_id.as_bytes()),
+            "The given account ID is invalid"
+        );
+        // Can only be called by a whitelisted factory or by the foundation.
+        if !self
+            .factory_whitelist
+            .contains(&env::predecessor_account_id())
+        {
+            self.assert_called_by_foundation();
+        }
+        self.whitelist.insert(&staking_pool_account_id)
+    }
+
+    /**************/
+    /* Foundation */
+    /**************/
+
+    /// Removes the given staking pool account ID from the whitelist.
+    /// Returns `true` if the staking pool was present in the whitelist before, `false` otherwise.
+    /// This method can only be called by the NEAR foundation.
+    pub fn remove_staking_pool(&mut self, staking_pool_account_id: AccountId) -> bool {
+        self.assert_called_by_foundation();
+        assert!(
+            env::is_valid_account_id(staking_pool_account_id.as_bytes()),
+            "The given account ID is invalid"
+        );
+        self.whitelist.remove(&staking_pool_account_id)
+    }
+
+    /// Adds the given staking pool factory contract account ID to the factory whitelist.
+    /// Returns `true` if the factory was not in the whitelist before, `false` otherwise.
+    /// This method can be called either by the NEAR foundation.
+    pub fn add_factory(&mut self, factory_account_id: AccountId) -> bool {
+        assert!(
+            env::is_valid_account_id(factory_account_id.as_bytes()),
+            "The given account ID is invalid"
+        );
+        self.assert_called_by_foundation();
+        self.factory_whitelist.insert(&factory_account_id)
+    }
+
+    /// Removes the given staking pool factory account ID from the factory whitelist.
+    /// Returns `true` if the factory was present in the whitelist before, `false` otherwise.
+    /// This method can only be called by the NEAR foundation.
+    pub fn remove_factory(&mut self, factory_account_id: AccountId) -> bool {
+        self.assert_called_by_foundation();
+        assert!(
+            env::is_valid_account_id(factory_account_id.as_bytes()),
+            "The given account ID is invalid"
+        );
+        self.factory_whitelist.remove(&factory_account_id)
+    }
+
+    /************/
+    /* Internal */
+    /************/
+
+    /// Internal method to verify the predecessor was the NEAR Foundation account ID.
+    fn assert_called_by_foundation(&self) {
+        assert_eq!(
+            &env::predecessor_account_id(),
+            &self.foundation_account_id,
+            "Can only be called by NEAR Foundation"
+        );
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use near_sdk::{testing_env, MockedBlockchain};
+
+    mod test_utils;
+    use test_utils::*;
+
+    #[test]
+    fn test_whitelist() {
+        let mut context = VMContextBuilder::new()
```

Comment:
how is it that we still didn't add this to the near-sdk-rs?

Reply:
`test_utils` 
![ehstoday_8410_success](https://user-images.githubusercontent.com/470453/82952212-6a27f580-9f5d-11ea-87d2-d6c4fe50a84d.png)